pandas/io/msgpack/_packer.pyx

```diff
@@ -234,7 +234,7 @@ cdef class Packer:
                 default_used = 1
                 continue
             else:
-                raise TypeError("can't serialize {thing!r}".format(thing=o))
+                raise TypeError(f"can't serialize {repr(o)}")
```

Comment:
Basically it's described in [pep-498](https://www.python.org/dev/peps/pep-0498/#s-r-and-a-are-redundant)(```!s, !r, and !a are redundant``` section).

Plus as someone who is new to python, at first I didn't understand the ```!r``` syntax. I do think that if we use ```repr()``` it will increase code readability (for new developers at least).

Reply:
fair enough but not seen this conversion in other PRs for #29547

to ensure consistency, could either update the issue with additional instructions or add a code check to disallow the redundant sytnax in f-strings (as a follow-on).

Reply:
Sure! love the idea of making a code style in the issue tab.

I don't have the confident to write this mini style guide, I really don't want to mislead people.

Reply:
@MomIsBestFriend ok if you'd make an issue about standarizing on repr(...) rather than ``!r`` agree that its more explicit; we would need to change any code and add a code-check.